features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiver.java

```diff
@@ -28,12 +28,189 @@
 
 package org.opennms.netmgt.trapd;
 
-/**
- * @author Seth
- */
-public interface TrapReceiver {
+import java.io.IOException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.util.Objects;
 
-    void start();
+import org.opennms.core.ipc.sink.api.AsyncDispatcher;
+import org.opennms.core.ipc.sink.api.MessageDispatcherFactory;
+import org.opennms.core.logging.Logging;
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.netmgt.config.TrapdConfig;
+import org.opennms.netmgt.config.trapd.TrapdConfiguration;
+import org.opennms.netmgt.dao.api.DistPollerDao;
+import org.opennms.netmgt.snmp.SnmpUtils;
+import org.opennms.netmgt.snmp.TrapInformation;
+import org.opennms.netmgt.snmp.TrapNotificationListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
-    void stop();
+public class TrapReceiver implements TrapNotificationListener, TrapReceiverService {
```

Comment:
handler vs receiver vs listener

In Syslog feature, we call the classes that open up the UDP socket and listen for packets listeners instead of receivers.